feat(Plugin): Add AlwaysExpandRoles

surgedevs opened the pull request:
Adds a plugin to always expand the role list in profile popouts.
This was highly requested and sparked drama in the Discord server, very small plugin, not sure what else to say.

Nuckyz commented:
We should probably migrate the settings from the  old ShowAllRoles

people who had it enabled will likely already want this enabled

surgedevs commented:
> We should probably migrate the settings from the old ShowAllRoles
> 
> people who had it enabled will likely already want this enabled

Added, also added the missing README.md

Sqaaakoi commented:
How about adding an option to hide the collapse button?

sadan4 commented:
> How about adding an option to hide the collapse button?

could you just use CSS for that?

Nuckyz commented on src/plugins/alwaysExpandRoles/README.md:
```diff
@@ -0,0 +1,3 @@
+# Always Expand Roles
+
+Always expands the role list in profile popouts, like the now deleted ShowAllRoles used to do
```
```suggestion
Always expands the role list in profile popouts
```

Nuckyz commented on src/plugins/alwaysExpandRoles/index.ts:
```diff
@@ -0,0 +1,37 @@
+    authors: [Devs.surgedevs],
+    patches: [
+        {
+            find: "action:\"EXPAND_ROLES\"",
```
```suggestion
            find: 'action:"EXPAND_ROLES"',
```

Nuckyz commented on src/plugins/alwaysExpandRoles/index.ts:
```diff
@@ -0,0 +1,37 @@
+        {
+            find: "action:\"EXPAND_ROLES\"",
+            replacement: {
+                match: /(\[\i,\i\]=\i\.useState\()!1\)/,
+                replace: "$1!0)"
```
```suggestion
                match: /(roles:\i(?=.+?(\i)\(!0\)[,;]\i\({action:"EXPAND_ROLES"}\)).+?\[\i,\2\]=\i\.useState\()!1\)/,
                replace: (_, rest) => `${rest}!0)`
```

Sorry if this regex looks like hell, it's just abusing lookaheads to make sure we are changing the correct useState

Sqaaakoi replied:
Why use a replacement function?

Nuckyz replied:
naming the capturing groups

Sqaaakoi replied:
Also that `\2` is almost certainly a typo

Nuckyz commented:
> How about adding an option to hide the collapse button?

not worth that's so minimal

Nuckyz commented:
Thank you!

surgedevs commented:
> Hello, thanks for the plugin!
> 
> I think adding an option to hide the collapse button would be good as part of the plugin. Since the roles are always expanded, there's not really a reason you would want to collapse them (since you expressed your preference not to collapse).

This was already discussed above.

```css
div:has(> div[class^=collapseButton_]) {
    display: none;
}
```

Nuckyz commented:
> Hello, thanks for the plugin!
> 
> I think adding an option to hide the collapse button would be good as part of the plugin. Since the roles are always expanded, there's not really a reason you would want to collapse them (since you expressed your preference not to collapse).

The button is so small it makes no difference to have it there or not. Also useful when you want to collapse for some reason
